Backend/Models/OrderEntity/Order.cs

```diff
@@ -66,33 +66,28 @@ public class Order : ISoftDeletable
 
         public Order() { }
 
-        public Order(
-            CreateOrderDTO dto,
-            string addressFrom,
-            string addressTo,
-            int pharmacyId,
-            IEnumerable<ProductBalance> productBalances
-        )
+        public Order(CreateOrderDTO dto, string addressFrom, string addressTo, int pharmacyId,
+            IEnumerable<ProductBalance> productBalances)
         {
             AddressFrom = addressFrom;
             AddressTo = addressTo;
             CreationDate = DateTime.Now;
             DeliveryDate = DetermineDeliveryDate(CreationDate);
             OrderStateId = OrderStateId.Created;
-            Total = CalculateTotalAmount(dto);
+            TotalSum = CalculateTotalAmount(productBalances);
             WarehouseId = dto.WarehouseId;
             PharmacyId = pharmacyId;
             OrderProductBalances = productBalances.Select(pb => new OrderProductBalance(this, pb)).ToList();
         }
 
-        public void UpdateFromDTO(CreateOrderDTO dto, IEnumerable<ProductBalance> productBalances)
+        public void UpdateFromDTO(List<ProductBalance> productBalances)
         {
```

Comment:
Why was this changed to List?

Reply:
This way we avoid ambiguity of possible double enumartion. Method body has a foreach loop which could be enumerated as both array or a list. If we specify `List` as a parameter instead of `IEnumerable`, whis ambiguity is removed.